lib/protest/runner.rb

```diff
@@ -42,6 +42,10 @@ def report(test)
     def fire_event(event, *args)
       event_handler_method = :"on_#{event}"
       @report.send(event_handler_method, *args) if @report.respond_to?(event_handler_method)
+
+      if Protest.fail_fast? && [:failure, :error].include?(event)
```

Comment:
> Do you prefer moving the logic there and adding both lines?

Yes, I'd prefer to have 2 exit statements in the right place instead of evaluating both conditions with every single event.

Thanks!